### ISSUE TITLE
Polish PET-MAD-DOS example

### ISSUE THREAD

**Issue by ceriottm** [lab-cosmo/atomistic-cookbook#257]
The PET-MAD-DOS example works, but is very clunky. Some of the issues require making a better interface at the `upet` level, but I list them here as the end goal is to get a nice, user-friendly recipe.

- [x] bug (?): 
```
from upet.calculator import PETMADDOSCalculator
import ase-> Kernel crashed
import ase
from upet.calculator import PETMADDOSCalculator-> fine
```

- [x] I get a bunch of Warning: `You are sending unauthenticated requests to the HF Hub. Please set a HF_TOKEN to enable higher rate limits and faster downloads`, do we not cache the checkpoint locally?
- [x] In general, all the helper functions should be part of the upet.dos package. Doesn't make sense (and it's not sustainable) to have a bunch of opaque magic functions distributed with the recipe, e.g.:
    - computing the energy grid and/or interpolating on a different grid
    - computing the alignment
    - computing the structure DOS and mask
- [x] Have a utility function in upet that pads a given DOS profile
- [x] Decrease learning rate for fine-tuning
- [x] use the same energy grid for plotting true and predicted DOS (meaning true dos has to be padded, meaning we need better helper functions)
- [x] remove large-ish data/ files, use another zenodo
- [x] consider whether to use convolution to speed up calculation of the alignment loss (probably out of scope here)

**Comment by HowWeiBin** [comment]
@Luthaf, regarding the `You are sending unauthenticated requests to the HF Hub. Please set a HF_TOKEN to enable higher rate limits and faster downloads`, I believe that upet doesn't cache the checkpoints locally and the first run always involves downloading the relevant models. The same issue is seen in https://atomistic-cookbook.org/examples/pet-mad/pet-mad.html for example. Do you know a way to fix this?

**Comment by HowWeiBin** [comment]
@pfebrer what do you mean by `Have a utility function in upet that pads a given DOS profile`, do you mean when preparing data for finetuning?

**Comment by Luthaf** [comment]
> Do you know a way to fix this?

Change what `upet` is doing? But yes, this is maybe not an issue for this repo ^^

**Comment by pfebrer** [comment]
> @pfebrer what do you mean by Have a utility function in upet that pads a given DOS profile, do you mean when preparing data for finetuning?

During the tutorial, there was a bit of confusion regarding the fact that the DOS needs to be padded to have a given number of grid points to do the finetuning. In the tutorial you hardcode something like `n_extra_points = 200`, which feels a bit like a magic number because in reality it depends on the number of grid points that PET produces and the number of grid points that the DOS profiles have. So we were thinking of having a `upet` function that accepts the raw DOS profiles and prepares them for training, doing the necessary padding.
